### DISCUSSION TITLE
Differences between Recognized `v` in cellData and `p.body.dataStream`

### DISCUSSION THREAD

**Discussion by IchiharaGaku** [dream-num/univer#2469] [Q&A | 常见问题] [answered]
I implemented a function to fetch added data upon button click using Univer Sheet on Next.js:

```javascript
const snapshot = activeWorkbook!.getSnapshot();
const sheetData = Object.values(snapshot.sheets).find((sheet) => {
  return sheet.id === sheetName;
});
```

When examining added data in a sheet, some data is retrieved as `value` while others are in `p.body.dataStream`, causing undefined issues when accessing `value`.

Question
In what cases is data recognized as `dataStream` instead of `value` ?

**Comment by hexf00** [comment, marked as answer]
When cell content contains rich text, it uses the `p` property to store the data. You can find more details here: https://univer.ai/guides/sheet/getting-started/cell-data#rich-text.
